### DISCUSSION TITLE
Bugs in MSS Configurator(2021.2, .3) while using REFCLK_x_PLL_NW

### DISCUSSION THREAD

**Discussion by henryding-emcore** [polarfire-soc/polarfire-soc-documentation#121] [Reference designs, FPGA and development kits] [answered]
In MSS, there is no option for me to choose MSS PLL reference clock source and RTC reference clock source to be the same source from NW PLL. Instead, seems REFCLK_0_PLL_NW is dedicated to MSS PLL and REFCLK_1_PLL_NW is dedicated to RTC. 
With this, the auto generated file ICICLE_MSS.v only exposes REFCLK_0_PLL_NW to the top and connects REFCLK_1_PLL_NW to VCC. 

![image](https://user-images.githubusercontent.com/98772058/177820663-74cb7bf9-48b2-4dc1-ab34-1d924691552d.png)

I have to manually modify my top file (generated from SmartDesign) and ICICLE_MSS.v file to connect REFCLK_1_PLL_NW to either OUT2 or OUT3 of NW PLL .

**Comment by hughbreslin** [comment, marked as answer]
Hi @henryding-emcore 

from the screenshot above, you aren't using DDR correct? I can reproduce this using the following MSS configuration:
![image](https://user-images.githubusercontent.com/44862881/177960855-1c4206d8-f53d-4b5d-9aec-27583f93fa5f.png)

If I enable DDR and use the REFCLK_1_PLL_NW for its reference clock I can see both reference clocks:
![image](https://user-images.githubusercontent.com/44862881/177961216-f018d069-cd3d-47f8-a6f1-7c1ea8b91719.png)

If I use the dedicated I/O from Bank5 as the reference clock the REFCLK_1_PLL_NW also doesn't show up

**Reply by henryding-emcore** [reply]
Hi Hugh, I'm glad you can reproduce the problem I saw. No, I'm not using DDR.

**Reply by hughbreslin** [reply]
Hi Henry, this is reported internally for a fix but it will be released in a future version of the configurator. 

For now the best suggestions are:
1. Do as you've done already and manually connect the pin
2. You could enable DDR so the pin is exposed and not include DDR support in your project so the DDR is never enabled / brought up

Thank you for reporting this
